scripts/openapi-convert/object-fixes.mjs

```diff
@@ -0,0 +1,47 @@
+"use strict";
+import { URL } from 'node:url';
+
+// Refactor `servers` field to be able to access `basePath` easily.
+function refactorServers(obj) {
+    if (!obj.servers) {
+        return;
+    }
+
+    let server = {
+        url: "",
+        variables: {
+            basePath: {
+                default: ""
+            }
+        }
+    }
+
+    const url = new URL(obj.servers[0].url);
+
+    if (obj.servers.length > 1) {
+        // In our case several URLs always mean both http and https for the same
+        // host.
+        server.variables.protocol = {
+            enum: ["http", "https"],
+            default: "https",
+        }
+        obj.servers.pop();
+
+        server.url = "{protocol}://" + url.host + "{basePath}";
+    } else {
+        server.url = url.origin + "{basePath}";
+    }
+
+    server.variables.basePath.default = url.pathname;
```

Comment:
I'm not really good at JavaScript but here's the idea that leaves key parts up for individual tweaking while having reasonable defaults applied across all definitions.
```suggestion
    let server = {
        url: "",
        variables: {
            protocol: {},
            hostname: {
                default: ""
            },
            basePath: {
                default: ""
            }
        }
    }

    const url = new URL(obj.servers[0].url);

    if (obj.servers.length > 1) {
        // In our case several URLs always mean both http and https for the same
        // host.
        obj.servers.pop()
    }

    server.url = "{protocol}://{hostname}{basePath}"
    server.variables.protocol = {
        enum: ["http", "https"],
        default: "https",
    }
    server.variables.hostname.default = url.host
    server.variables.basePath.default = url.pathname
```